include/swift/SIL/SILBuilder.h

```diff
@@ -1510,15 +1593,25 @@ class SILBuilder {
 
   StructExtractInst *createStructExtract(SILLocation Loc, SILValue Operand,
                                          VarDecl *Field, SILType ResultTy) {
-    return insert(new (getModule()) StructExtractInst(
-        getSILDebugLocation(Loc), Operand, Field, ResultTy));
+    return createStructExtract(Loc, Operand, Field, ResultTy,
+                               Operand.getOwnershipKind());
   }
 
   StructExtractInst *createStructExtract(SILLocation Loc, SILValue Operand,
                                          VarDecl *Field) {
     auto type = Operand->getType().getFieldType(Field, getModule(),
                                                 getTypeExpansionContext());
-    return createStructExtract(Loc, Operand, Field, type);
+    return createStructExtract(Loc, Operand, Field, type,
+                               Operand.getOwnershipKind());
+  }
+
+  StructExtractInst *
+  createStructExtract(SILLocation Loc, SILValue Operand, VarDecl *Field,
+                      SILType ResultTy,
+                      ValueOwnershipKind forwardingOwnershipKind) {
+    return insert(new (getModule()) StructExtractInst(
```

Comment:
I am a little nervous about struct_extract/tuple_extract. I am worried this may let them take an owned value by mistake. My thought: add a check in SILVerifier itself since this is a sort of structural test on StructExtract, TupleExtract. That I am cool.